### DISCUSSION TITLE
Installation on Windows Server 2022

### DISCUSSION THREAD

**Discussion by Indirectelex** [microsoft/terminal#13983] [Q&A] [answered]
Wonder if this is possible to install the latest version manually on windows server 2022, if so please can you tell here how to do it?

**Comment by RoyiAvital** [comment, marked as answer]
I managed to do so by:

1. Download the latest package of Windows Terminal for Windows 10 (Currently `Microsoft.WindowsTerminal_Win10_1.15.2524.0_8wekyb3d8bbwe.msixbundle`).
2. Download the latest VC UWP RunTime (`Microsoft.VCLibs.x64.14.00.Desktop.appx`). See [C++ Runtime framework packages for Desktop Bridge](https://learn.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge).  For some reason the download page [Microsoft Visual C++ UWP Desktop Runtime Package](https://www.microsoft.com/en-us/download/details.aspx?id=102159) doesn't work. It downloads a `txt` file.

Then open a PowerShell window (Not in Terminal though) and do:

1. `Add-AppxPackage -Path <PathTo>\Microsoft.VCLibs.x64.14.00.Desktop.appx`.
2. `Add-AppxPackage -Path <PathTo>\Microsoft.WindowsTerminal_Win10_1.15.2524.0_8wekyb3d8bbwe.msixbundle`.

**Reply by DHowett** [reply]
You can also get the requisite dependency packages from the "preinstallation kit" ZIP file on the assets page! :smile:

**Reply by RoyiAvital** [reply]
@DHowett , Could you ping someone at MS about the issue I raised on the [Microsoft Visual C++ UWP Desktop Runtime Package](https://www.microsoft.com/en-us/download/details.aspx?id=102159) page.

**Reply by bentman** [reply]
Thanks! helped with this script to install Dev Tools on Windows Server 2022 - 
[Add-DevToMyWinServer.ps1](https://gist.github.com/bentman/638c478ae791598780c70749139e382f#file-add-devtomywinserver-ps1)

**Reply by SkewelVsAll** [reply]
Does this add the Desktop/Windows Explorer context menu entries too? And will the same work on Windows 10 LTSC?

**Reply by scarolan** [reply]
@RoyiAvital you are my hero, thank you for this

**Comment by jozefizso** [comment]
I tried the `choco upgrade -y microsoft-windows-terminal` command to install the Terminal on Windows Server 2022 (10.0.20348.1547) and also the manual steps above.

I use an user account with admin rights (not the built-in Administrator account) to develop on Windows Server and the Terminal was not opening for me.

Solution was to run the Terminal as administrator for the first time. I can run it as standard user afterwards.

**Comment by wsy** [comment]
Just a remind:
As of today (Sep17th, 2023), [`Windows Terminal v1.17.11461.0`](https://github.com/microsoft/terminal/releases/tag/v1.17.11461.0) requires [`Microsoft.UI.Xaml.2.7`](https://github.com/microsoft/microsoft-ui-xaml/releases/tag/v2.7.3)
I manually downloaded that package and installed [`Microsoft.UI.Xaml.2.7`](https://github.com/microsoft/microsoft-ui-xaml/releases/tag/v2.7.3) and I was able to install [`Windows Terminal v1.17.11461.0`](https://github.com/microsoft/terminal/releases/tag/v1.17.11461.0).
Also, version `2.8` of `Microsoft.UI.Xaml` won't work with `Windows Terminal v1.17.11461.0`.

**Reply by tylamb1** [reply]
# ################################################################################################
# Install Windows Terminal on Server but really anything
# ################################################################################################

Invoke-WebRequest -Uri https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -outfile Microsoft.VCLibs.x86.14.00.Desktop.appx
Add-AppxPackage Microsoft.VCLibs.x86.14.00.Desktop.appx

# https://github.com/microsoft/microsoft-ui-xaml/releases/
# https://github.com/microsoft/microsoft-ui-xaml/releases/tag/v2.8.5
Invoke-WebRequest -Uri https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.5/Microsoft.UI.Xaml.2.8.x64.appx -outfile Microsoft.UI.Xaml.2.8.x64.appx
Add-AppxPackage Microsoft.UI.Xaml.2.8.x64.appx

# install WindowsTerminal
# get the latest version and update the links below
# https://github.com/microsoft/terminal/releases
Invoke-WebRequest -Uri https://github.com/microsoft/terminal/releases/download/v1.18.2822.0/Microsoft.WindowsTerminal_1.18.2822.0_8wekyb3d8bbwe.msixbundle -outfile Microsoft.WindowsTerminal_1.18.2822.0_8wekyb3d8bbwe.msixbundle
Add-AppxPackage Microsoft.WindowsTerminal_1.18.2822.0_8wekyb3d8bbwe.msixbundle

**Reply by davetapley** [reply]
I put this in a script and made it an `.exe` with [PS2EXE](https://github.com/MScholtes/PS2EXE), for easier install:
https://github.com/JEFuller/install-windows-terminal/releases/tag/v1.0.0

**Reply by ahpooch** [reply]
If someone is interested in script that could install latest available Windows Terminal or Windows Terminal Preview for Current user or All users scope, capable to perform Offline Installations - feel free to use this repository: https://github.com/ahpooch/Install-WindowsTerminal

**Comment by AndreyRyzhakov** [comment]
Is it possible to install on windows terminal server for all users?

**Reply by wsy** [reply]
Maybe this link will help: https://superuser.com/questions/647927/install-windows-store-app-package-appx-for-all-users

**Reply by ahpooch** [reply]
Install for All users scope could be done with this script if you want.
https://github.com/ahpooch/Install-WindowsTerminal

**Comment by cheema-corellian** [comment]
For me, the simplest option was to use scoop package manager to install the "windows-terminal" package.  It takes care of everything for you and installs the latest windows terminal and any of its dependencies.

You can get scoop from: https://scoop.sh/

**Reply by SkewelVsAll** [reply]
Windows Terminal on Scoop has its own problems, the WT on there always gives me an error on start-up saying that I'm missing a font whereas the MS Store variant does not.

**Reply by jessiewestlake** [reply]
I'm not sure if this is the problem you were facing, but it might just be complaining about not having Cascadia Code. You could find and install it from here: https://github.com/microsoft/cascadia-code/releases/latest

Background info: https://learn.microsoft.com/en-us/windows/terminal/cascadia-code

**Comment by ahpooch** [comment]
If someone is interested in script that could install latest available Windows Terminal or Windows Terminal Preview for Current user or All users scope, capable to perform Offline Installations - feel free to use this repository: https://github.com/ahpooch/Install-WindowsTerminal
